### PR TITLE
3d tiles renderer js migration

### PR DESCRIPTION
**Pull request by mgermerie**
## Description

Migrate towards [3DTilesRendererJS](https://github.com/NASA-AMMOS/3DTilesRendererJS) for 3D tiles. This PR is a draft from a branch initiated by @jailln on which all work related to 3D tiles migration should be merged before migration is complete and this PR can be merged.

## Motivation and Context

#2225

## Remaining TODO

See the full list [here](https://github.com/iTowns/itowns/issues/2225#issuecomment-1999238765)

### CODE REVIEW

**Comment by Desplandis** [comment]
Rebased on `master` (commit https://github.com/iTowns/itowns/commit/d091207146600f80dfd755c5616a16d9333718b0)

**Review by jailln — APPROVED** [review]

File: src/Layer/OGC3DTilesLayer.js [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+        if (model.isPoints) {
+            const pointsMaterial = new PointsMaterial({
+                mode: this.pntsMode,
+                shape: this.pntsShape,
+                classificationScheme: this.classification,
+                sizeMode: this.pntsSizeMode,
+                minAttenuatedSize: this.pntsMinAttenuatedSize,
+                maxAttenuatedSize: this.pntsMaxAttenuatedSize,
+            });
+            pointsMaterial.copy(material);
```
Comment (airnez) [inline src/Layer/OGC3DTilesLayer.js:275]: This overwrites `pointsMaterial.userData` and thus removes `pointsMaterial.userData.needTransparency = {}` initialized at     `PointsMaterial` construction. This breaks conditional transparency on classified point clouds.
I suggest modifying [this method](https://github.com/iTowns/itowns/blob/cb410cd2f6f7d0562473bd04d1ccd18ecbd42312/src/Renderer/PointsMaterial.js#L284-L300). What do you think? We might add something like that before copying : 
```
const needTransparency = source.userData.needTransparency !== undefined ? source.userdata.needTransparency
        : this.userData.needTransparency;
```
and then apply again the property after copying : 
```
this.userData.needTransparency = needTransparency;
```

Comment (airnez) [reply src/Layer/OGC3DTilesLayer.js:275]: @jailln @Desplandis

Comment (jailln) [reply src/Layer/OGC3DTilesLayer.js:275]: good catch, thanks ! I implemented what you suggested

**Review by Desplandis — APPROVED** [review]
Looks good to me!

**Pull request merged**